src/pages/proposal-commons/proposal-details/proposal-details.tsx

```diff
@@ -79,9 +78,14 @@ interface RouterParameters {
 const ProposalDetailsPage = () => {
   useScrollToTop();
   const { typeAndVoteId } = useParams<RouterParameters>();
+  const { push } = useHistory();
   const decoded = useMemo(() => decodeProposalTypeAndVoteId(typeAndVoteId), [typeAndVoteId]);
 
-  if (!decoded) return <NotFoundPage />;
+  if (!decoded) {
+    push('/404');
```

Comment:
Pretty sure we want to keep the same behaviour. That is, to stay on the same URL but render the Not Found page instead of navigating to another URL. FYI we don't have `/404` route.

Reply:
Hmm... I thought we may want to adapt to the new 404 page, which is more than just a `Not Found` text now. It`s a page on its own, that has a layout around it. 

I see your point, thought. But if we want to keep the URL `not-found.tsx` needs to be split and to export two components: content with layout and content without layout. The `proposal-details.tsx` doesn't need a layout, it already has one.

But this approach seems a bit unusual to me. Do we really want to do it? I feels like the way to go is redirecting everything that is not found to `/404` and NotFound page. This would solve your final comment too, with the missing route. Wdyt?

<img width="833" alt="image" src="https://github.com/user-attachments/assets/14d087ca-5335-483b-82a7-5361d237f849">

Reply:
Or maybe we just want a Not Found text in there, instead of the entire 404 page content?

Reply:
So this is the `<ProposalDetailsPage />` component:
```tsx
          <Route path="/governance/:typeAndVoteId" exact>
            <ProposalDetailsPage />
          </Route>
```
It returns the whole page (including the layout), so you can return your `<NotFoundPage />` (which will have its own layout).

Reply:
Oh, I see now. Not sure where I messed up that it put layout inside layout for me during development.
Fixed